apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/Tasks/TestScheduler.cs

```diff
@@ -334,10 +334,47 @@ public override void Wait(Task task)
 
             public override async Task WhenAll(IEnumerable<Task> tasks)
             {
+                var exceptions = new List<Exception>();
+                var cancelled = false;
                 foreach (var task in tasks)
                 {
                     await this.ConfigureAwaitHideErrors(task);
+                    switch (task.Status)
+                    {
+                        case TaskStatus.RanToCompletion:
+                            break;
+                        case TaskStatus.Faulted:
+                            exceptions.AddRange(task.Exception.InnerExceptions);
+                            break;
+                        case TaskStatus.Canceled:
+                            cancelled = true;
```

Comment:
this will loop and await every task even if the first cancels which appears to result in the same result (a TaskCanceledException).
Seems it might be better to do a WaitAny, loop thru the completed/final tasks removing them from the list, bail if canceled, and then continue with the rest until empty...?